aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -628,6 +630,43 @@ public class AwsProperties implements Serializable {
    */
   public static final String LAKE_FORMATION_DB_NAME = "lakeformation.db-name";
 
+  /** Region to be used by the SigV4 protocol for signing requests. */
+  public static final String REST_SIGV4_SIGNER_REGION = "rest.sigv4.signing-region";
+
+  /** The service name to be used by the SigV4 protocol for signing requests. */
+  public static final String REST_SIGV4_SIGNING_NAME = "rest.sigv4.signing-name";
+
+  /** The default service name (API Gateway and lambda) used during SigV4 signing. */
+  public static final String REST_SIGV4_SIGNING_NAME_DEFAULT = "execute-api";
+
+  /**
+   * Configure the static access key ID used for SigV4 signing.
+   *
+   * <p>When set, the default client factory will use the basic or session credentials provided
+   * instead of reading the default credential chain to create S3 access credentials. If {@link
+   * #REST_SIGV4_SESSION_TOKEN} is set, session credential is used, otherwise basic credential is
+   * used.
+   */
+  public static final String REST_SIGV4_ACCESS_KEY_ID = "rest.sigv4.access-key-id";
```

Comment:
Do we need both `rest` and `sigv4` prefixes? 

If we removed "rest" then this would appear more generic because it isn't clear what SigV4 connections this applies to. We should probably keep `rest` then.

If we remove "sigv4" then we would have `rest.access-key-id`, `rest.session-token`, `rest.sigv4-enabled`, etc. Those seem fairly reasonable to me... but I'm definitely not sure they belong in `AwsProperties`.

Reply:
I think the issue here is that there are multiple aws keys used and just using `rest` isn't clear that it's for request signing vs fileio (which has an `s3` prefix).

Reply:
There are also a lot of other AWS services that are configurable, so I think we want to be pretty explicit about the fact that these credentials are used only for this specific purpose.

Reply:
Isn't `rest` a specific enough purpose?

If the concern is that this conflicts with other properties in this class, then let's move them. These are more associated with REST than with AWS.

Reply:
I've updated the property names

Reply:
`rest.` also sounds good to me!